src/circuit/gadgets/constraints.rs

```diff
@@ -1,13 +1,53 @@
 // Initially taken from: rust-fil-proofs/storage-proofs-core/src/gadgets/
 
+use crate::circuit::add_clause_single;
+use crate::circuit::gadgets::case::multi_case_aux;
+use crate::circuit::gadgets::case::CaseClause;
+use crate::circuit::gadgets::data::GlobalAllocations;
```

Comment:
We should not be using `GlobalAllocations` here, so this import is an indication that something needs to be restructured.

Reply:
The same goes (but less emphatically) for the imports from `case`.

Reply:
solved

Reply:
I still see the import. What am I missing?

Reply:
Sorry, I missed this during rebase. Fixed now.